### PR TITLE
Allow applets to specify the platform(s) they support (X11/Wayland)

### PR DESCRIPTION
**Pull request by wmww**
* Add a `Platforms` list to the applet metadata files
* Fail to load an applet when on a platform it doesn't support (logs a warning and shows the same error dialog to the user as with other applet errors)
* In the addto window, grey out and make unselectable applets that don't support the current platform

Wayland support still isn't here. To test, I recommend changing the platform value of an applet to be Wayland only, and noting how it is then not loaded or addable on X11.

### CODE REVIEW

File: mate-panel/panel-addto.c [hunk 1]
```diff
@@ -681,7 +695,12 @@ static void panel_addto_make_application_model(PanelAddtoDialog* dialog)
 	if (dialog->filter_application_model != NULL)
 		return;
 
-	store = gtk_tree_store_new(NUMBER_COLUMNS, G_TYPE_STRING, G_TYPE_STRING, G_TYPE_POINTER, G_TYPE_STRING);
+	store = gtk_tree_store_new(NUMBER_COLUMNS,
```
Comment (raveit65) [inline mate-panel/panel-addto.c]: Any chance of using a space before a bracket, same like in the rest of the file?
I see that at several places, i don't want to mark every occurrence in your PR, please fix.

**Comment by raveit65** [comment]
> I recommend changing the platform value of an applet to be Wayland only,

How can i do that?

**Comment by raveit65** [comment]
Ok, i guess in need to remove X11 from this line to see that the fish won't load under X11?
```
[root@mother rave]# cat /usr/share/mate-panel/applets/org.mate.panel.FishApplet.mate-panel-applet | grep Wayland
Platforms=X11;Wayland;
```

**Comment by raveit65** [comment]
Nice, this works as expected with fish and clock applet under X11 :)
`Platforms=Wayland;`
Both applets are grey-out in add-to-panel dialog.

**Comment by wmww** [comment]
Formatting fixed. This PR adds two string (`Not compatible with Wayland` and `Not compatible with X11`) that should be translated. I wrapped them in `_(...)`, but I don't know what else if anything I ought to do.

**Comment by raveit65** [comment]
After merging we can update transifex resource file (mate-panel.pot) to catch new translation strings, np.

File: mate-panel/panel-addto.c (hunk 1 above)
Comment (raveit65) [reply mate-panel/panel-addto.c]: Looks much better, thanks.

**Pull request merged**